cclib/parser/gaussianparser.py

```diff
@@ -1853,46 +1857,86 @@ def natural_orbital_single_spin_parsing(inputfile, updateprogress_title):
         #     2  C    0.002063
         # ...
         #
-        if line[1:25] == "Mulliken atomic charges:" or line[1:18] == "Mulliken charges:" or \
-           line[1:23] == "Lowdin Atomic Charges:" or line[1:16] == "Lowdin charges:" or \
-           line[1:37] == "Mulliken charges and spin densities:" or \
-           line[1:32] == "Mulliken atomic spin densities:":
-
-            has_spin = 'spin densities' in line 
-            has_charges = 'charges' in line 
-
-            if has_charges and not hasattr(self, "atomcharges"):
-                self.atomcharges = {}
-
-            if has_spin and not hasattr(self, "atomspins"):
-                self.atomspins = {}
-
-            ones = next(inputfile)
-
-            charges = []
-            spins = []
-            nline = next(inputfile)
-            while not "Sum of" in nline:
-                if has_charges:
-                    charges.append(float(nline.split()[2]))
-
-                if has_spin and has_charges:
-                    spins.append(float(nline.split()[3]))
-
-                if has_spin and not has_charges:
-                    spins.append(float(nline.split()[2]))
-
-                nline = next(inputfile)
-
-            if "Mulliken" in line:
-                if has_charges:
-                    self.atomcharges["mulliken"] = charges
-                if has_spin:
-                    self.atomspins["mulliken"] = spins
+        # Spins may be included in the same section, e.g.
+        #
+        # Mulliken charges and spin densities:
+        #               1          2
+        #     1  O   -0.596188   0.019133
+        #     2  C    0.320624   0.000869
+        #
+        # APT and Lowdin charges are also displayed in this way
+        def extract_charges_spins(prop,header):
```

Comment:
In order to make things clearer especially when it comes to function please add type annotations and docstrings. In this case it should look like this I guess:
```
def extract_charges_spins(prop: str, header: str) -> str:
    """This function does this thing.
    
    Inputs:
        prop - short explanation what prop is ideally one line
        header - short explanation what header is ideally one line
    """

Reply:
obviously str in the type annotations should be the right type :)

Reply:
have added documentation here